spec/mailers/user_mailer_spec.rb

```diff
@@ -50,6 +52,43 @@
     allow(Setting).to receive(:default_language).and_return('en')
   end
 
+  shared_examples_for 'mail is sent' do
+    it 'actually sends a mail' do
+      expect(ActionMailer::Base.deliveries.size).to eql(1)
+    end
+
+    it 'is sent to the recipient' do
+      expect(ActionMailer::Base.deliveries.first.to).to include(recipient.mail)
```

Comment:
You don't need single, you just need to assure that the recipients in the deliveries include the ``recipient.mail``. To make another C# reference: I'd suggest a lambda which maps each delivery to their respective recipients on which you can then look for the email address in question.